diff --git a/blocks.def.h b/blocks.def.h
index 9c22d68..fc8ae7a 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -9,3 +9,15 @@ static const Block blocks[] = {
 //sets delimeter between status commands. NULL character ('\0') means no delimeter.
 static char delim[] = " | ";
 static unsigned int delimLen = 5;
+
+// Button roles definition
+// left click  - 1
+// wheel click - 2
+// right click - 3
+// scroll up   - 4
+// scroll down - 5
+static const char* buttonAction = "1";
+static const char* buttonTerminate = "2";
+static const char* buttonInfo = "3";
+static const char* buttonScrollUp = "4";
+static const char* buttonScrollDown = "5";
diff --git a/dwmblocks.c b/dwmblocks.c
index 5500189..3176237 100644
--- a/dwmblocks.c
+++ b/dwmblocks.c
@@ -69,10 +69,15 @@ void getcmd(const Block *block, char *output, const char *button)
 	const char formatString[] =
 		"/bin/env "
 		"BUTTON=%s "
+		"BUTTON_ACTION=%s "
+		"BUTTON_TERMINATE=%s "
+		"BUTTON_INFO=%s "
+		"BUTTON_SCROLL_UP=%s "
+		"BUTTON_SCROLL_DOWN=%s "
 		"%s";
 	const size_t commandSize = sizeof(formatString) + strlen(block->command); // should be more than enough
 	char command[commandSize];
-	sprintf(command, formatString, buttonValue, block->command);
+	sprintf(command, formatString, buttonValue, buttonAction, buttonTerminate, buttonInfo, buttonScrollUp, buttonScrollDown, block->command);
 	FILE *cmdf = popen(command, "r");
 	if (!cmdf)
 		return;
