diff --git a/dwmblocks.c b/dwmblocks.c
index 13ec863..7d35e84 100644
--- a/dwmblocks.c
+++ b/dwmblocks.c
@@ -3,6 +3,7 @@
 #include<string.h>
 #include<unistd.h>
 #include<signal.h>
+#include <time.h>
 #ifndef NO_X
 #include<X11/Xlib.h>
 #endif
@@ -60,6 +61,8 @@ static int returnStatus = 0;
 //opens process *cmd and stores output in *output
 void getcmd(const Block *block, char *output)
 {
+	nanosleep((const struct timespec[]){{0, 20000000L}}, NULL);
+
 	if (block->signal)
 	{
 		output[0] = block->signal;
