diff --git a/blocks.def.h b/blocks.def.h
index 9c22d68..5bdca50 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -1,11 +1,15 @@
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
-	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
-	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
-	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
+	/*Icon*/	 /*Command*/		                                      /*Update Interval*/	/*Update Signal*/
+	{"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/memory.sh",       2,                    10},
+	{"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/uptime.sh",       120,                  16},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/volume.sh",       1,                    12},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/checkupdates.sh", 3600,                 13},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/date.sh",         60,                   14},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/tray.sh",         60,                   11},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/dwmblocks/warnings.sh",     10,                   15},
 };
 
 //sets delimeter between status commands. NULL character ('\0') means no delimeter.
-static char delim[] = " | ";
+static char delim[] = "";
 static unsigned int delimLen = 5;
