diff --git a/blocks.def.h b/blocks.def.h
index 9c22d68..134d9cb 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -1,11 +1,18 @@
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
-	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
-	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
-	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
+    /*Icon*/	 /*Command*/		                                          /*Update Interval*/   /*Update Signal*/
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 0 10 0 1 memory.sh",     2,                    10},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 1 10 0 1 uptime.sh",     120,                  16},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 2 10 0 1 volume.sh",     1,                    12},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 3 10 0 3 music_prev.sh", 0,                    17},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 4 10 1 3 music.sh",      3,                    18},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 5 10 2 3 music_next.sh", 0,                    19},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 6 10 0 1 packages.sh",   3600,                 13},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 7 10 0 1 date.sh",       60,                   14},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 8 10 0 1 tray.sh",       60,                   11},
+    {"",         "$LINUX_SETUP_ROOT/steps/dwm/status_bar.sh 9 10 0 1 warnings.sh",   10,                   15},
 };
 
 //sets delimeter between status commands. NULL character ('\0') means no delimeter.
-static char delim[] = " | ";
+static char delim[] = "";
 static unsigned int delimLen = 5;
