diff --git a/x.c b/x.c
index c8bdc94..7d77356 100644
--- a/x.c
+++ b/x.c
@@ -259,6 +259,7 @@ static double usedfontsize = 0;
 static double defaultfontsize = 0;
 
 static char *opt_alpha = NULL;
+static char *opt_background = NULL;
 static char *opt_class = NULL;
 static char **opt_cmd  = NULL;
 static char *opt_embed = NULL;
@@ -796,6 +797,9 @@ xloadcols(void)
 	static int loaded;
 	Color *cp;
 
+	if (opt_background)
+		strcpy(colorname[defaultbg], opt_background);
+
 	if (loaded) {
 		for (cp = dc.col; cp < &dc.col[dc.collen]; ++cp)
 			XftColorFree(xw.dpy, xw.vis, xw.cmap, cp);
@@ -2095,6 +2099,9 @@ main(int argc, char *argv[])
 	case 'A':
 		opt_alpha = EARGF(usage());
 		break;
+	case 'B':
+		opt_background = EARGF(usage());
+		break;
 	case 'c':
 		opt_class = EARGF(usage());
 		break;
